examples/ca-opportunity-example/src/ca_common_grants/utils/date_transform.py

```diff
@@ -0,0 +1,207 @@
+"""
+Date transformation utilities for the CA Grants Portal data transformer.
+
+This module provides functionality to transform various date formats into
+standardized datetime objects.
+"""
+
+import re
+from datetime import datetime, timezone
+from enum import Enum
+
+
+class DateFormat(Enum):
+    """Enum for date format types."""
+
+    LONG = "9999-12-31 23:59:59"
+    SHORT = "12/31/99"
+
+
+def transform_date(date_str: str, output_format: DateFormat) -> datetime:
+    """
+    Transform a date string to a datetime object.
+
+    Args:
+        date_str: The date string to transform
+        output_format: The desired output format (SHORT or LONG)
+
+    Returns:
+        A datetime object in the specified format
+
+    Raises:
+        ValueError: If the date string is in an unrecognized format
+
+    """
```

Comment:
Future consideration: Because this kind of transformation seems like it would be used consistently across implementations, I'm wonder if this would be worth including in the the SDK itself.